fearless/Modules/Staking/StakingMain/StakingMainPresenter.swift

```diff
@@ -211,13 +245,27 @@ extension StakingMainPresenter: StakingMainPresenterProtocol {
     }
 
     func performRedeemAction() {
+        guard let view = view else { return }
+        let selectedLocale = view.localizationManager?.selectedLocale
+        guard controllerAccount != nil else {
+            let baseState = stateMachine.viewState(using: { (state: BaseStashNextState) in state })
+            wireframe.presentMissingController(
+                from: view,
+                address: baseState?.stashItem.controller ?? "",
+                locale: selectedLocale
+            )
+            return
+        }
         wireframe.showRedeem(from: view)
     }
 }
 
 extension StakingMainPresenter: StakingStateMachineDelegate {
-    func stateMachineDidChangeState(_: StakingStateMachineProtocol) {
+    func stateMachineDidChangeState(_ stateMachine: StakingStateMachineProtocol) {
         provideState()
+        if let stashItem = stateMachine.viewState(using: { (state: BondedState) in state.stashItem }) {
```

Comment:
This won't work for all cases, for example when we are nominator. I would rather bind the logic when we receive stash item. Ideally, I would store stashAccount and controllerAccount in state.

Reply:
oops, this is redundant code, i've removed those lines. Currently fetchController calls on stashItem
<img width="446" alt="Screenshot 2021-05-20 at 14 03 08" src="https://user-images.githubusercontent.com/17689921/118968213-4e8adb80-b974-11eb-9a05-39a59feffcf3.png">